### ISSUE TITLE
Extracting 7z SFX files fails on linux

### ISSUE THREAD

**Issue by chrisvigil** [th3w1zard1/KOTORModSync#35]
Extracting 7z SFX files files fails on Linux. 

When attempting to install [Improved AI](https://deadlystream.com/files/file/833-improved-ai/) from the KOTOR2_Full.toml list instead of extracting `Modified AI.exe` in step 2 it attempts to open it in my default web browser.

My workaround was to extract `Modified AI.exe` with the 7z CLI application, repack it as a regular 7z and modify the install instructions.

KOTORModSync Version 1.0.2
OS: Arch Linux

**Comment by th3w1zard1** [comment]
Yes, this is a known bug but it seems I've forgotten to post about it. Thanks for the reminder.

Realistically there's nothing I can do, the self-executable extractor is a Windows binary and won't open on Unix systems.

I have tried SharpCompress and tried building 7zip and packaging their .so/.dll/.dylib libs but those won't open self-extracting executables either.

You said the 7z cli works on linux for .exe sfx's? Could you show the exact commands you ran? If so I could package the 7z cli with kotormodsync's unix releases.

**Comment by chrisvigil** [comment]
It the same command as extracting any other 7z file: 
``` 7z x Modified\ AI.exe ```

I did some quick research now and Arch apparently used [p7zip](https://github.com/p7zip-project/p7zip) as it's 7z implementation which is provided by the `p7zip-full` package on Ubuntu. There is also the [official](https://github.com/ip7z/7zip) 7z implementation (proved by `7zip` on ubuntu) which i confirmed can also extract sfx's in the same way.

**Comment by th3w1zard1** [comment]
So what I'm understanding is there's no singlecli binary I can package with ModSync that'll work on all of these different unix systems? It sounds like I'll need to do a few uname -s checks to determine what operating system the user is using and then use their package manager to install 7zip.

**Comment by chrisvigil** [comment]
No you should be fine including a binary for all linux systems and another for macOS The statically linked binary included with  the [official version](https://github.com/ip7z/7zip/releases/tag/23.01) should work best I think.

Either that or check for an installed binary and use that, both implementations appear to work the same way.

**Comment by ProjectSynchro** [comment]
Looking around to see if this can be easily done, I was reminded that Homebrew supports both macOS and Linux nowadays.

Looking at their 'sevenzip' package script: https://github.com/Homebrew/homebrew-core/blob/e9dd2b385f7a4c2b1b2fe6cf09469939df38f585/Formula/s/sevenzip.rb

It looks like they download upstream releases and use them there. The Linux x64 release is statically built, and the macOS one is a statically built universal binary.

See: 
https://7-zip.org/a/7z2403-linux-x64.tar.xz
https://7-zip.org/a/7z2403-mac.tar.xz

**Comment by ProjectSynchro** [comment]
Been thinking about solving this problem when I get the chance, I looked around for platform agnostic managed implementations for C# and there is a lack of any good ones (or any at all). 

It may make the most sense to tear out SharpCompress in favour of helper functions to do the archive decompression using the 7z binary itself.

On Linux and macOS we could look in the PATH for 7z first and then fallback to a bundled 7z binary. On Windows including a 7z binary is fairly trivial.

On Linux and macOS you can decompress sfx archives without problems using the standalone 7z binary.
